### PR TITLE
refactor dates calculation

### PR DESCRIPTION
**Pull request by Juan-EBI**
- refactor time procesor
- minor improvements

### CODE REVIEW

**Comment by jhoanmanuelms** [comment]
LTGM. Just some typos.

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/mapping/SubmissionDbMapper.kt [hunk 1]
```diff
@@ -34,30 +41,78 @@ class SubmissionDbMapper {
...
+     */
+    private fun <T : Tabular, S, U> toEitherList(
+        elements: List<T>,
+        transform: (T) -> S,
+        tableBuilder: (List<S>) -> U
+    ):
+        MutableList<Either<S, U>> {
+
+        val map = elements
+            .groupBy { it.tableIndex != NO_TABLE_INDEX }
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/mapping/SubmissionDbMapper.kt]: Wow, I hadn't really read this code. This is really powerful 😲

File: submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/PropertiesProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package ac.uk.ebi.biostd.submission.processors
+
+import ebi.ac.uk.model.ExtendedSubmission
+import ebi.ac.uk.persistence.PersistenceContext
+import java.util.UUID
+
+/**
+ * Calculate and set general purpose properties values as submission title and submission secret.
+ */
+class PropertiesProcessor : SubmissionProcessor {
```
Comment (jhoanmanuelms) [inline submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/PropertiesProcessor.kt]: 👍

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt [hunk 3]
```diff
@@ -5,31 +5,44 @@ import ebi.ac.uk.model.Submission
 import ebi.ac.uk.model.constants.SubFields
 import java.time.Instant
 
+/**
+ * Contains accessor for attributes that has specific meaning in page tab specification.
+ */
+
+/**
+ * Indicate that submission should be register for specific project.
```
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt]: one comment about this one,  this property should not be used into the code to identify the submission project is just a keyword interpreted at submission time.  We need strongly relation between submission and project to have an strong relation.

**Pull request merged**